urld558a3d988_gao715108023_resourceMonitor_tgz-pJ8-com_sinopec_agent_AgentJ8/src/com/sinopec/cache/HistroyMsgCache.java

```diff
@@ -0,0 +1,523 @@
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created with IntelliJ IDEA. User: gaochuanjun Date: 13-10-24 Time: 下午2:58 To
+ * change this template use File | Settings | File Templates.
+ */
+public class HistroyMsgCache {
+
+	private static final Log LOG = LogFactory.getLog(HistroyMsgCache.class);
+
+	/**
+	 * 具体内容存放的地方
+	 */
+	ConcurrentHashMap<String, ConcurrentHashMap<String, DynamicInfoSerializable>> cache;
+
+	private ScheduledExecutorService scheduleService;
+
+	/**
+	 * 存储数据的时间间隔，默认为30秒
+	 */
+	private int storeInterval = 30;
+
+	private JDBCCommon jdbcCommon;
+
+	private String dbType;
+
+	private String dataBaseServerIP;
+
+	public HistroyMsgCache() {
+		init();
+	}
+
+	public HistroyMsgCache(int storeInterval, JDBCCommon jdbcCommon, String dbType, String dataBaseServerIP) {
+		this.storeInterval = storeInterval;
+		this.jdbcCommon = jdbcCommon;
+		this.dbType = dbType;
+		this.dataBaseServerIP = dataBaseServerIP;
+		init();
+	}
+
+	private void init() {
+		cache = new ConcurrentHashMap<String, ConcurrentHashMap<String, DynamicInfoSerializable>>();
+		scheduleService = Executors.newScheduledThreadPool(1);
+		scheduleService.scheduleAtFixedRate(new StoreHisMsgDataSchedule(cache, jdbcCommon, dbType), 0, storeInterval, TimeUnit.SECONDS);
+		if (LOG.isInfoEnabled()) {
+			LOG.info("The service of storing history data has started!");
+		}
+	}
+
+	class StoreHisMsgDataSchedule implements Runnable {
+
+		ConcurrentHashMap<String, ConcurrentHashMap<String, DynamicInfoSerializable>> cache;
+
+		private Connection conn;
+
+		private JDBCCommon jdbcCommon;
+
+		private String dbType;
+
+		private volatile boolean storeData;
+
+		private PreparedStatement pstmtOne;
+
+		private PreparedStatement pstmtThree;
+
+		StoreHisMsgDataSchedule(ConcurrentHashMap<String, ConcurrentHashMap<String, DynamicInfoSerializable>> cache, JDBCCommon jdbcCommon, String dbType) {
+			this.cache = cache;
+			this.jdbcCommon = jdbcCommon;
+			this.dbType = dbType;
+			setStoreData(false);
+
+		}
+
+		private void initialOracleDB() throws SQLException {
+			conn = jdbcCommon.getOrclConn();// 与Oracle数据库创建连接
+			if (conn == null) {
+				LOG.info("Can't connect to Oracle-->" + jdbcCommon.getIpAddress());
+			} else {
+				conn.setAutoCommit(false);
+				pstmtOne = conn.prepareStatement(OracleSQL.insertComputeOneMonthSQL);
+				pstmtThree = conn.prepareStatement(OracleSQL.insertComputeThreeMonthSQL);
+				setStoreData(true);
+			}
+		}
+
+		private void initialMySQLDB() throws SQLException {
+			conn = jdbcCommon.getMySQLConn(dataBaseServerIP);// 与MySQL数据库创建连接
+			if (this.conn == null) {
+				LOG.info("Can't connect to History-MySQL-->" + dataBaseServerIP);
+			} else {
+				conn.setAutoCommit(false);
+				pstmtOne = conn.prepareStatement(MySQLSQL.insertComputeOneMonthSQL);
+				pstmtThree = conn.prepareStatement(MySQLSQL.insertComputeThreeMonthSQL);
+				setStoreData(true);
+			}
+		}
+
+		public void insertComputeNodeHistoryAvgData() throws SQLException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 107 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.